services/github/github-checks-status.service.js

```diff
@@ -0,0 +1,99 @@
+'use strict'
+
+const Joi = require('joi')
+const { NotFound, InvalidParameter } = require('..')
+const { GithubAuthV3Service } = require('./github-auth-service')
+const { documentation, errorMessagesFor } = require('./github-helpers')
+
+const schema = Joi.object({
+  state: Joi.equal('failure', 'pending', 'success'),
+}).required()
+
+module.exports = class GithubChecksStatus extends GithubAuthV3Service {
+  static category = 'build'
+  static route = {
+    base: 'github/checks-status',
+    pattern: ':user/:repo/:ref',
+  }
+
+  static examples = [
+    {
+      title: 'GitHub branch checks state',
+      namedParams: {
+        user: 'badges',
+        repo: 'shields',
+        ref: 'master',
+      },
+      staticPreview: this.render({
+        state: 'success',
+      }),
+      keywords: ['status'],
+      documentation,
+    },
+    {
+      title: 'GitHub commit checks state',
+      namedParams: {
+        user: 'badges',
+        repo: 'shields',
+        ref: '91b108d4b7359b2f8794a4614c11cb1157dc9fff',
+      },
+      staticPreview: this.render({
+        state: 'success',
+      }),
+      keywords: ['status'],
+      documentation,
+    },
+    {
+      title: 'GitHub tag checks state',
+      namedParams: {
+        user: 'badges',
+        repo: 'shields',
+        ref: '3.3.0',
+      },
+      staticPreview: this.render({
+        state: 'success',
+      }),
+      keywords: ['status'],
+      documentation,
+    },
+  ]
+
+  static defaultBadgeData = { label: 'checks' }
+
+  static render({ state }) {
+    if (state === 'success') {
+      return {
+        message: `passing`,
+        color: 'brightgreen',
+      }
+    } else if (state === 'pending') {
+      return {
+        message: `pending`,
+        color: 'yellow',
+      }
+    } else {
+      return {
+        message: `failing`,
+        color: 'red',
+      }
+    }
+  }
+
+  async handle({ user, repo, ref }) {
+    let state
+    try {
+      ;({ state } = await this._requestJson({
+        url: `/repos/${user}/${repo}/commits/${ref}/status`,
+        errorMessages: errorMessagesFor('ref not found'),
+        schema,
+      }))
+    } catch (e) {
+      if (e instanceof NotFound) {
+        throw new InvalidParameter({ prettyMessage: 'invalid ref' })
+      }
+      throw e
+    }
+
+    return this.constructor.render({ state })
```

Comment:
We can simplify this a bit. The `errorMessagesFor` helper is specifying the error message explicitly for the not found scenario, so we can just specify whatever value we want there instead of putting one value that's caught (and essentially tossed) and then mapped.

In the case of a 404 with the GitHub badges, we don't really have a great way of knowing whether the issues is with the owner/repo params or the noun param (the ref in this cases), so for the not found case the majority of our GitHub badges have an error like `noun or repo not found` and we'll want to do the same here.

We should also utilize the render helper as well:

```suggestion
    let { state } = await this._requestJson({
      url: `/repos/${user}/${repo}/commits/${ref}/status`,
      errorMessages: errorMessagesFor('ref or repo not found'),
      schema,
    })

    return renderBuildStatusBadge({ status: state })
```